_posts/2021-10-09-software-and-projects-as-systems.md

```diff
@@ -0,0 +1,191 @@
+- the expected deliverables: goals, approval criteria, operation environment like integration interfaces, hosting or data platforms and so on
+
+### Organization and stakeholders
+
+Implementing a change using some technical approach or another might not always seem very different from a software developer's point of view. Impacts on project stakeholders may, however, be very different. Differences might not be obvious if a developer spends only a short time in a project, if a project is a Proof of Concept (PoC) project or if the project manager is overly protective.
+
+The following describes typical stakeholders for a software project:
+
+- Business representatives
+  - Project delivery is typically expected to affect business areas by
+    - revenue changes
+    - cost changes
+  - Business stakeholders who don't collaborate with the project or with each other are one reason for unexpected impacts.
+- End users
+- IT and operating teams
+  - Delivery must fit into the existing IT and software environment when it is deployed. Delivery is typically operated and maintained by a different team than the one creating it.
+- User support
+  - Corporate or consumer End user support
+  - Business-to-business support
+- Marketing
+  - Internal for corporate users or external for public communications
+- Training
+  - End user organizations and people may require scheduling additional training or existing training content must be updated.
+- Organizational entities
+  - For example, the managers responsible for resources
+- External or internal collaborators
+  - A delivery may be a joint creation by different delivery teams or companies.
+- Regulations and legal
+  - Delivery may have to comply with national or international regulations. Compliancy could be ensured by collaboration with a domain expert.
+  - In some domains delivery may be a pioneering pilot project whose results will affect upcoming domain standards.
+- IPR
+  - Is delivery respecting Intellectual Property Rights (IPR) like previous patents and licensing terms?
+    - A common software developer challenge is evaluating software licenses of software components and tools.
+  - Will any new IPR be created for the delivery or customer organizations?
+- Open-source community
+  - Project may affect existing open-source projects or create new ones
+  - Open-source projects affect software projects, for example if the delivery uses software components which become unreliable or not maintained.
+
+## Final words
+
+The everyday challenge for many software engineers is to remind themselves to take the step back and demand the big picture. A common pitfall is when a customer or end user comes to the table with a ready solution. It is tempting to take that for granted and trust that "someone" has already asked the "what if" questions. The presented thoughts are not groundbreaking, but hopefully a good reminder.
```

Comment:
"A common pitfall is when a customer or end user comes to the table with a ready solution. " tämä on erittäin hyvä lause ja huomio.